src/Mod/PartDesign/App/FeatureSketchBased.cpp

```diff
@@ -666,6 +705,154 @@ void ProfileBased::generatePrism(TopoDS_Shape& prism,
     }
 }
 
+void ProfileBased::generateTaperedPrism(const TopoDS_Shape& sketchshape,
+                                        const gp_Dir& direction,
+                                        const double Length,
+                                        const double Length2,
+                                        const double angle,
+                                        const double angle2,
+                                        std::list<TopoDS_Shape>& drafts)
+{
+    // inspect sketchshape
+    // that it is not null was already checked
+    TopoDS_Wire sourceWire;
+    if (sketchshape.ShapeType() == TopAbs_WIRE) {
+        ShapeFix_Wire aFix;
+        aFix.Load(TopoDS::Wire(sketchshape));
+        aFix.FixReorder();
+        aFix.FixConnected();
+        aFix.FixClosed();
+        sourceWire = aFix.Wire();
+    }
+    else if (sketchshape.ShapeType() == TopAbs_FACE) {
+        TopoDS_Wire outerWire = ShapeAnalysis::OuterWire(TopoDS::Face(sketchshape));
+        sourceWire = outerWire;
+    }
+    else if (sketchshape.ShapeType() == TopAbs_COMPOUND) {
+        TopoDS_Iterator it(sketchshape);
+        for (; it.More(); it.Next()) {
+            generateTaperedPrism(it.Value(), direction, Length, Length2, angle, angle2, drafts);
+        }
+    }
+    else {
+        throw Base::RuntimeError("Only a wire or a face is supported");
+    }
+
+    if (sourceWire.IsNull()) {
+        throw Base::RuntimeError("Offset for base shape could not made");
+    }
+
+    // calculate lateral offset and vectors for top and bottom shape of the loft
+    double lateralOffset = tan(angle * M_PI / 180.0) * Length;

```

Comment:
I propose `Base::toRadians(angle)` instead of `angle * M_PI / 180.0`

Reply:
Thanks, I will change this.
At first the PR must compile for the CI.

Reply:
done now